infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -502,6 +502,14 @@ type ServerConfigs struct {
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
 }
 
+// RegistryServerConfigs creates a registry server for the feast service, with specified container configurations.
+type RegistryServerConfigs struct {
+	ServerConfigs `json:",inline"`
+	// RestAPIEnabled determines if the registry should serve using the REST API instead of gRPC.
+	// +kubebuilder:default=true
+	RestAPIEnabled bool `json:"restAPIEnabled,omitempty"`
```

Comment:
Does it make more sense for the field to match the flag/option name? "Enabled" is implited by setting to `true`. Also, this field should not be set by default as this is not the default behaviour of the registry's remote server. This flag should be intentionally set by the user as it is an "option" in the CLI.
```suggestion
	// RestAPI determines if the registry should serve using the REST API instead of gRPC.
	RestAPI bool `json:"restAPI,omitempty"`
```
One more note on this, we don't use kubebuilder to set defaults. If this was a feild where a default should be set, the current approach/design with this CR is to set that default in `status.applied`.

So, if the intention was to force this as a default setting (i think some broader discussions should happen first if so), you'd want to make the field a pointer bool `*bool` and handle the default in status.

Reply:
Most of the operators (like kubeflow has EnableIstio/MetricsCollectorSpec.Enabled) follows Enabled in their naming convention for boolean fields so added enabled earlier, removed Enabled from field now in https://github.com/feast-dev/feast/pull/5364/commits/66fd234f724ca4265dbb70b37962f495b8cc04d2

I do want to apply some default settings like when ui is deployed, registry server served with rest api, but I will do it another PR and agree that can be discussed separately.

Reply:
> I do want to apply some default settings like when ui is deployed, registry server served with rest api

it's my understanding that the feast UI has no reliance on the registry remote server (grpc or rest). isn't the UI server a standalone component of feast? all it currently needs, as i understand it, is a valid `feature_store.yaml` with proper data store configs.